src/nrsc5.c

```diff
@@ -702,14 +702,24 @@ void nrsc5_report_lost_sync(nrsc5_t *st)
     nrsc5_report(st, &evt);
 }
 
-void nrsc5_report_hdc(nrsc5_t *st, unsigned int program, const uint8_t *data, size_t count)
+void nrsc5_report_hdc(nrsc5_t *st, unsigned int program, const packet_t* pkt)
 {
     nrsc5_event_t evt;
 
     evt.event = NRSC5_EVENT_HDC;
     evt.hdc.program = program;
-    evt.hdc.data = data;
-    evt.hdc.count = count;
+    evt.hdc.data = NULL;
+    evt.hdc.count = 0;
+    evt.hdc.flags = 0;
+
+    if (pkt->shape == PACKET_FULL)
+    {
+        evt.hdc.data = pkt->data;
+        evt.hdc.count = pkt->size;
+    }
+    if (pkt->flags & PACKET_FLAG_CRC_ERROR)
+        evt.hdc.flags |= NRSC5_PKT_FLAGS_CRC_ERROR;
```

Comment:
I assume the separation here is so that we could add flags in the future that are only used for internal bookkeeping inside the library?

Reply:
Yes. I do this by habit. I could switch to using NRSC5_PKT_FLAGS_CRC_ERROR internally within the library if you want me too.